helix-term/src/commands/typed.rs

```diff
@@ -321,6 +321,24 @@ fn buffer_previous(
     Ok(())
 }
 
+fn buffer_nth(cx: &mut compositor::Context, args: Args, event: PromptEvent) -> anyhow::Result<()> {
+    if event != PromptEvent::Validate {
+        return Ok(());
+    }
+    let n: usize = args[0]
+        .parse()
+        .map_err(|_| anyhow!("provided argument is not an integer"))?;
+    ensure!(n != 0);
+    let (id, _) = if args.has_flag("reverse") {
+        cx.editor.documents.iter().nth_back(n - 1)
+    } else {
+        cx.editor.documents.iter().nth(n - 1)
+    }
+    .ok_or(anyhow!("buffer {n} is out of range"))?;
```

Comment:
```suggestion
    .ok_or_else(|| anyhow!("buffer {n} is out of range"))?;
```

using `ok_or` here will always evaluate the `anyhow!` macro call, even when there is no error